### PR TITLE
[Docs] Update CLI page formatting

### LINKED ISSUE #545: [DOCS] Update CLI page formatting closer to models
**Is your feature request related to a problem? Please describe.**

The [CLI page ](https://docs.unitycatalog.io/usage/cli/) should be closer to the [Models page](https://docs.unitycatalog.io/usage/models/) making use of:
* More admonitions, i.e. notes / warnings / tips
Screenshot 2024-10-01 at 21 04 09

* More graphics/screen GIFs showing how the CLI works scubas this [MLflow animated GIF](https://docs.unitycatalog.io/assets/images/mlflow-unitycatalog-model-view-2.0.gif)

* More annotations

Screenshot 2024-10-01 at 21 03 51

**Describe the solution you would like**

As noted in the problem area, please provide more extensive documentation for the CLI. 

**Additional context**

### PR DESCRIPTION
**Pull request by avriiil**
This PR updates the formatting of the CLI docs page to be more like the Models page. Will close #545.

Signed-off-by: Avril Aysha <68642378+avriiil@users.noreply.github.com>

### CODE REVIEW

**Comment by HorizonNet** [comment]
@avriiil Before going into detail, are you sure that this should link to #515?

**Comment by avriiil** [comment]
@HorizonNet - good catch, should be #545, I've corrected it.

File: docs/usage/cli.md [hunk 1]
```diff
@@ -1,535 +1,724 @@
...
-bin/uc catalog list [--max_results <max_results>]
+bin/uc catalog list \
+  [--max_results <max_results>] # (1)
+```
+
+1. `max_results`: optional flag to set the maximum number of results to return.
+
+This will output:
+
+```sh
```
Comment (HorizonNet) [inline docs/usage/cli.md:39]: Use `console` instead for console outputs.

File: docs/usage/cli.md [hunk 2]
```diff
@@ -1,535 +1,724 @@
...
+  --name <name> # (1)
 ```
 
-- `name`: The name of the catalog.
+1. `name`: The name of the catalog.
 
-### 1.3 Create a Catalog
+This should output:
 
 ```sh
```
Comment (HorizonNet) [inline docs/usage/cli.md:60]: Use `console` instead for console outputs.

File: docs/usage/cli.md [hunk 3]
```diff
@@ -1,535 +1,724 @@
...
-## 6. Registered model and model version management
+You can use Unity Catalog with MLflow to govern and access your ML and AI models. Read more
 
 Please refer to [MLflow documentation](https://mlflow.org/docs/latest/index.html) to learn how to use MLflow to create,
 register, update, use, and delete registered models and model versions.
 
-## 7. CLI Server Configuration
+You can list the registered models in your UC namespace using:
+
+```bash title="List registered models"
```
Comment (HorizonNet) [inline docs/usage/cli.md:595]: Use `sh` instead to stay consistent. Also applies to the other code snippets below.

**Review by HorizonNet — CHANGES_REQUESTED** [review]
Left a few comments regarding code snippet formatting. Rest LGTM.

**Review by HorizonNet — APPROVED** [review]
LGTM

**Review by dennyglee — APPROVED** [review]
Oh, this looks great - thanks!

**Pull request merged**